app/controllers/api/groups_controller.rb

```
@@ -161,6 +161,86 @@ def update_marks
+        render 'shared/http_status', locals: { code: '200', message:
+            HttpStatusHelper::ERROR_CODE['message']['200'] }, status: 200
+      rescue
+        # Some error occurred
+        render 'shared/http_status', locals: { code: '500', message:
+            extra_mark.errors.full_messages.first }, status: 500
+      end
+    end
+
+    def remove_extra_marks
+      assignment = Assignment.find(params[:assignment_id])
+      if assignment.nil?
+        render 'shared/http_status', locals: { code: '404', message:
+            'No assignment exists with that id' }, status: 404
+        return
+      end
+
+      group = Group.find(params[:id])
+      if group.nil?
+        render 'shared/http_status', locals: { code: '404', message:
+            'No group exists with that id' }, status: 404
+        return
+      end
+      if group.grouping_for_assignment(params[:assignment_id])
+              .has_submission?
+        result = group.grouping_for_assignment(params[:assignment_id])
+                      .current_submission_used
+                      .get_latest_result
+      else
+        render 'shared/http_status', locals: { code: '404', message:
+            'No submissions exist for that group' }, status: 404
+        return
+      end
+      extra_mark = ExtraMark.find_by(result_id: result.id,
+                                     description: params[:description],
+                                     extra_mark: params[:extra_marks])
+      if extra_mark.nil?
+        render 'shared/http_status', locals: { code: '404', message:
+            'No such Extra Mark exist for that result' }, status: 404
+      end
+      begin
+        extra_mark.destroy
+        result.update_total_mark
+        # Successfully deleted the Extra Mark; render success
+        render 'shared/http_status', locals: { code: '200', message:
+            HttpStatusHelper::ERROR_CODE['message']['200'] }, status: 200
+      rescue ActiveRecord::RecordNotDestroyed
```

please report the error that occurred as the message